File: Easy/2026-03-27-2946-Matrix-Similarity-After-Cyclic-Shifts/solution.py
class Solution:
    def shiftRow(self, row, k, direction):
        n = len(row)
        k = k % n
        if direction == 'left':
            return row[k:] + row[:k]
        else:
            return row[-k:] + row[:-k]

    def matrixSimilar(self, mat: List[List[int]], k: int) -> bool:
        original = [row[:] for row in mat]
        for i in range(k):
            for j in range(len(mat)):
                if j % 2 == 0:
                    mat[j] = self.shiftRow(mat[j], 1, 'left')
                else:
                    mat[j] = self.shiftRow(mat[j], 1, 'right')
        return mat == original

File: Easy/2026-03-27-2946-Matrix-Similarity-After-Cyclic-Shifts/solution.java
class Solution {
    public boolean canBeEqual(int[][] mat, int k) {
        int m = mat.length;
        int n = mat[0].length;

        // Normalize k to be within the bounds of the row length
        k = k % n;

        for (int i = 0; i < m; i++) {
            if (i % 2 == 0) {
                // Even indexed rows shift left
                if (!isEqual(mat[i], shiftLeft(mat[i], k))) {
                    return false;
                }
            } else {
                // Odd indexed rows shift right
                if (!isEqual(mat[i], shiftRight(mat[i], k))) {
                    return false;
                }
            }
        }
        return true;
    }

    private int[] shiftLeft(int[] row, int k) {
        int n = row.length;
        int[] shifted = new int[n];
        for (int i = 0; i < n; i++) {
            shifted[i] = row[(i + k) % n];
        }
        return shifted;
    }

    private int[] shiftRight(int[] row, int k) {
        int n = row.length;
        int[] shifted = new int[n];
        for (int i = 0; i < n; i++) {
            shifted[i] = row[(i - k + n) % n];
        }
        return shifted;
    }

    private boolean isEqual(int[] row1, int[] row2) {
        for (int i = 0; i < row1.length; i++) {
            if (row1[i] != row2[i]) {
                return false;
            }
        }
        return true;
    }
}